src/core/receiver/control_thread.cc

```diff
@@ -92,23 +92,37 @@ ControlThread::ControlThread()
     const std::string empty_implementation;
     std::string src_impl = configuration_->property("SignalSource.implementation", empty_implementation);
     int src_count = configuration_->property("Receiver.sources_count", 1);
-    if (src_impl.empty() && (src_count != 1))
+    if (src_impl.empty())
         {
-            int num_src = 0;
-            for (int i = 0; i < src_count; i++)
+            if ((src_count == 1))
                 {
-                    std::string src_impl_multiple = configuration_->property("SignalSource" + std::to_string(i) + ".implementation", empty_implementation);
-                    if (!src_impl_multiple.empty())
-                        {
-                            num_src++;
-                        }
+                    conf_has_signal_sources_ = false;
                 }
-            if (num_src != src_count)
+            else
                 {
-                    src_impl = std::string("");
+                    int num_src = 0;
+                    for (int i = 0; i < src_count; i++)
+                        {
+                            std::string src_impl_multiple = configuration_->property("SignalSource" + std::to_string(i) + ".implementation", empty_implementation);
+                            if (!src_impl_multiple.empty())
+                                {
+                                    num_src++;
+                                }
+                        }
+                    if (num_src == src_count)
+                        {
+                            conf_has_signal_sources_ = true;
+                        }
+                    else
+                        {
+                            conf_has_signal_sources_ = false;
+                        }
```

Comment:
This whole section can be simplified, by taking advantage of the defined `bool`->`int` conversion:
```c++
            for (auto i = 0; i < src_count; ++i)
                 {
                     auto src_impl_multiple = configuration_->property("SignalSource" + std::to_string(i) + ".implementation", empty_implementation);
                     num_src += !src_impl_multiple.empty();
                 }
             conf_has_signal_sources_ = (num_src == src_count);
```
although I question whether this config file validation is of any value at all, since the flowgraph will fail if the config is mal-formed.